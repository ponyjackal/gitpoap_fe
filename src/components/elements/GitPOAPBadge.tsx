import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { TextLight, MidnightBlue } from '../../colors';

type Props = {
  imgUrl: string;
};

type HexProps = {
  size: number;
  borderSize: number;
};

const Hexagon = styled.div`
  clip-path: polygon(5% 25%, 50% 0, 95% 25%, 95% 75%, 50% 100%, 5% 75%);
`;

const HexOuterBorder = styled(Hexagon)<HexProps>`
  position: relative;
  --s: ${(props) => rem(props.size + 4 * props.borderSize)};
  width: var(--s);
  height: calc(var(--s) * 1);
  background-color: ${TextLight};
`;
const HexInnerBorder = styled(Hexagon)<HexProps>`
  --s: ${(props) => rem(props.size + 2 * props.borderSize)};
  position: absolute;
  top: ${(props) => rem(props.borderSize)};
  left: ${(props) => rem(props.borderSize)};

  width: var(--s);
  height: calc(var(--s) * 1);
  background: ${MidnightBlue};
`;

const HexBadge = styled(Hexagon)<{ url: string } & HexProps>`
  --s: ${(props) => rem(props.size)};
  position: absolute;
  top: ${(props) => rem(props.borderSize)};
  left: ${(props) => rem(props.borderSize)};

  width: var(--s);
  height: calc(var(--s) * 1);
  display: inline-block;
  font-size: initial; /* we reset the font-size if we want to add some content */
  background: url('${(props) => props.url}') no-repeat center center;
  background-size: cover;
`;

export const GitPOAPBadge = ({ imgUrl }: Props) => {
  const size = 200;
  const borderSize = 4;
  return (
    <HexOuterBorder size={size} borderSize={borderSize}>
      <HexInnerBorder size={size} borderSize={borderSize}>
        <HexBadge url={imgUrl} size={size} borderSize={borderSize} />
      </HexInnerBorder>
    </HexOuterBorder>
  );
};

/* Another option is to use SVG ~ https://stackoverflow.com/questions/67458234/mask-image-with-svg-shape-and-add-a-border*/
