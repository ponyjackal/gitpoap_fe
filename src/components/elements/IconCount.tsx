import styled from 'styled-components';
import { rem } from 'polished';
import { ExtraHover, ExtraPressed, TextGray, TextLight } from '../../colors';

type Props = {
  icon: React.ReactNode;
  href?: string;
  count: number;
};

const Count = styled.span`
  font-family: PT Mono;
  font-style: normal;
  font-weight: bold;
  font-size: ${rem(12)};
  line-height: ${rem(17)};
  letter-spacing: ${rem(0.5)};
  text-transform: uppercase;
  color: ${TextLight};
  margin-left: ${rem(6)};
`;

const Container = styled.a`
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;

  path {
    transition: 200ms fill ease;
    fill: ${TextGray};
  }

  &:hover {
    path {
      fill: ${ExtraHover};
    }
  }
  &:active {
    path {
      fill: ${ExtraPressed};
    }
  }
`;

export const IconCount = ({ icon, href, count }: Props) => {
  return (
    <Container href={href}>
      {icon}
      {count && <Count>{count}</Count>}
    </Container>
  );
};
