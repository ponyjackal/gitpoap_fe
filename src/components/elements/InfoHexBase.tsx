import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { BackgroundPanel } from '../../colors';

type Props = {
  children: React.ReactNode;
};

const StyledInfoHex = styled.div`
  display: inline-flex;
  flex-direction: column;
  min-width: ${rem(250)};
`;

const UpperTriangle = styled.div`
  clip-path: polygon(50% 0%, 0% 100%, 100% 100%);
  width: 100%;
  height: ${rem(70)};
  background-color: ${BackgroundPanel};
`;

const LowerTriangle = styled(UpperTriangle)`
  clip-path: polygon(50% 100%, 0 0, 100% 0);
`;

const Body = styled.div`
  background-color: ${BackgroundPanel};
  min-height: ${rem(100)};
`;

export const InfoHexBase = (props: Props) => {
  return (
    <StyledInfoHex>
      <UpperTriangle />
      <Body>{props.children}</Body>
      <LowerTriangle />
    </StyledInfoHex>
  );
};
