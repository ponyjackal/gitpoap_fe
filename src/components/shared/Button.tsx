import React from 'react';
import styled from 'styled-components';
import { Button as MButton } from '@mantine/core';
import { rem } from 'polished';
import { PrimaryBlue, TextAccent } from '../../colors';

type Props = {
  children: React.ReactNode;
  className?: string;
  onClick?: React.MouseEventHandler;
  disabled?: boolean;
  leftIcon?: React.ComponentProps<typeof MButton>['leftIcon'];
};

const StyledButton = styled(MButton)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background-color: ${PrimaryBlue};
  font-family: 'PT Mono', monospace;
  font-weight: 700;
  color: white;
  text-transform: uppercase;
  font-size: ${rem(12)};
  line-height: ${rem(18)};
  letter-spacing: ${rem(2)};
  transition: 200ms background ease;
  border-radius: ${rem(6)};
  padding: ${rem(8)} ${rem(14)};
  height: auto;

  &:hover {
    box-shadow: none;
    background-color: ${TextAccent};
  }

  &:active {
    background-color: #20457e;
  }
`;

export const Button = ({ children, className, onClick, disabled, leftIcon }: Props) => {
  return (
    <StyledButton
      compact
      className={className}
      size="md"
      disabled={disabled}
      onClick={onClick}
      leftIcon={leftIcon}
    >
      {children}
    </StyledButton>
  );
};
