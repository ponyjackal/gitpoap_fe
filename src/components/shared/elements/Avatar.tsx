import styled from 'styled-components';
import { rem } from 'polished';

export const Avatar = styled.img`
  height: ${rem(80)};
  width: ${rem(80)};
  border-radius: 50%;
`;
