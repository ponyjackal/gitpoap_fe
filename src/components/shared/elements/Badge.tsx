import styled from 'styled-components';
import { rem } from 'polished';
import { BackgroundPanel3, TextLight } from '../../../colors';

export const Badge = styled.div`
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: ${rem(0.5)} ${rem(4)};
  background: ${BackgroundPanel3};
  border-radius: ${rem(4)};

  font-family: PT Mono;
  font-style: normal;
  font-weight: bold;
  font-size: ${rem(11)};
  line-height: ${rem(18)};
  letter-spacing: ${rem(1.2)};
  text-transform: uppercase;
  color: ${TextLight};
`;
