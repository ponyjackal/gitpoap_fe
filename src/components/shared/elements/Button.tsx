import { Button as ButtonUI, ButtonProps } from '@mantine/core';
import React from 'react';

type Props = ButtonProps & {
  className?: string;
  onClick?: React.MouseEventHandler;
  onMouseEnter?: React.MouseEventHandler;
  onMouseLeave?: React.MouseEventHandler;
  id?: string;
};

export const Button = (props: Props) => {
  return <ButtonUI {...props}>{props.children}</ButtonUI>;
};
