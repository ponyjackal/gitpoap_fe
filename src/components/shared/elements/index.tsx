export * from './Avatar';
export * from './Badge';
export * from './Button';
export * from './Checkbox';
export * from './ClaimCircle';
export * from './CollapsibleAddress';
export * from './CopyableText';
export * from './DateInput';
export * from './Divider';
export * from './GitPOAPBadge';
export * from './Header';
export * from './HexagonPath';
export * from './IconCount';
export * from './InfoHexBase';
export * from './Input';
export * from './InputWrapper';
export * from './Label';
export * from './Loader';
export * from './MultiSelect';
export * from './NavLink';
export * from './NumberInput';
export * from './OrgName';
export * from './Pagination';
export * from './POAPBadge';
export * from './Radio';
export * from './RelativeDate';
export * from './Select';
export * from './Share';
export * from './Skeletons';
export * from './Text';
export * from './TextArea';
export * from './Title';
export * from './TwitterShareButton';
