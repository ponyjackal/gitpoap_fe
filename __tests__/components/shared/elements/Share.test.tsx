import { render } from '@testing-library/react';
import 'jest-styled-components';
import { Share } from '../../../../src/components/shared/elements';

describe('Share', () => {
  it('renders a Share', () => {
    const { container } = render(<Share textToCopy="textToCopy" />);
    const share = container.firstChild;

    expect(share).toBeInTheDocument();
    expect(share).toMatchSnapshot();
  });
});
