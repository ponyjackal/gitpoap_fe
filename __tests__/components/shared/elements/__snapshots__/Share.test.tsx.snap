// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share renders a Share 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  font-size: 0.75rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.125rem;
  -moz-letter-spacing: 0.125rem;
  -ms-letter-spacing: 0.125rem;
  letter-spacing: 0.125rem;
  text-transform: uppercase;
  color: #838495;
  -webkit-transition: 150ms color ease;
  transition: 150ms color ease;
  -webkit-transition: 150ms color ease;
  transition: 150ms color ease;
  cursor: pointer;
}

.c1 > * {
  color: #838495;
  -webkit-transition: 150ms color ease;
  transition: 150ms color ease;
}

.c1 > * {
  -webkit-transition: 150ms color ease;
  transition: 150ms color ease;
}

.c1:hover:not([disabled]) {
  color: #5596F6;
}

.c1:hover:not([disabled]) > * {
  color: #5596F6;
}

.c1:active:not([disabled]) {
  color: #1761CF;
}

.c1:active:not([disabled]) > * {
  color: #1761CF;
}

.c1[disabled] {
  color: #393A51;
}

.c1[disabled] > * {
  color: #393A51;
}

.c2 {
  margin-left: 0.3125rem;
}

<div
  class="c0"
>
  <div
    class="c1 emotion-0"
  >
    Share
    <svg
      class="c2"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 12v8a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-8"
      />
      <polyline
        points="16 6 12 2 8 6"
      />
      <line
        x1="12"
        x2="12"
        y1="2"
        y2="15"
      />
    </svg>
  </div>
</div>
`;
